### ISSUE TITLE
Multi-task with HMGNN

### ISSUE THREAD

**Issue by kypss93034** [shuix007/HMGNN#3]
Hi, I read your paper. Thank you, it looks so helpful for me.
But one part of code is difficult to me.
In your code, it's look for single task predict. 
But, how can I used it for the multi-task train and predict?

**Comment by shuix007** [comment]
Hi,

Thank you for your interest in HMGNN. I will update the code to accommodate multi-task prediction soon.

Currently, you can modify the output module in [this file](https://github.com/shuix007/HMGNN/blob/124f0ad13a228b2210f7602f7cea04c9dec97883/Model/HMGNN.py#L96) by replacing all the 1s with the number of tasks that you need. You also need to make sure the size of the mean and the std are consistent with your prediction.

You need to change attn_score = F.softmax(h, dim=1) in [this line](https://github.com/shuix007/HMGNN/blob/124f0ad13a228b2210f7602f7cea04c9dec97883/Model/HMGNN.py#L156) to attn_score = F.softmax(h, dim=1).unsqueeze(-1)

Then you need to make a minor change in [this line](https://github.com/shuix007/HMGNN/blob/124f0ad13a228b2210f7602f7cea04c9dec97883/Model/HMGNN.py#L230) to change torch.cat with torch.stack

The training and evaluation functions should be able to handle multi-task when the batched data y is of shape (n, k) where k is the number of your tasks.

Thank you,
Zeren Shui

**Comment by kypss93034** [comment]
Thanks for your reply. 
Apart from the above-mentioned changes, also needed to be changed in the main.py files or not?
Looking forward to the updated code about multi-task prediction.
Thank you
